src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -0,0 +1,198 @@
+  }
+
+  @Override
+  public String getStaticLangReturnTypeName(Method method, MethodConfig methodConfig) {
+    return converter.getTypeName(method.getOutputType()).getFullName();
+  }
+
+  @Override
+  public List<String> getDocLines(ProtoElement element) {
+    if (!(element instanceof Method)) {
+      return super.getDocLines(element);
+    }
+    Method method = (Method) element;
+    String text = DocumentationUtil.getDescription(method);
+    text = lowerFirstLetter(text);
+    return super.getDocLines(getApiMethodName(method) + " " + text);
+  }
+
+  @Override
+  public String getAndSavePagedResponseTypeName(ModelTypeTable typeTable, TypeRef resourceType) {
+    String typeName = converter.getTypeNameForElementType(resourceType).getNickname();
+    int dotIndex = typeName.indexOf('.');
+    if (dotIndex >= 0) {
+      typeName = typeName.substring(dotIndex + 1);
+    }
+    return Name.anyCamel(typeName).join("iterator").toUpperCamel();
+  }
+
+  private static String lowerFirstLetter(String s) {
+    if (s.length() > 0) {
+      s = Character.toLowerCase(s.charAt(0)) + s.substring(1);
+    }
+    return s;
+  }
+
+  private static String upperFirstLetter(String s) {
+    if (s.length() > 0) {
+      s = Character.toUpperCase(s.charAt(0)) + s.substring(1);
+    }
+    return s;
+  }
+
+  @Override
+  public String getGrpcClientTypeName(Interface service) {
+    // getNickname() thinks the client is a message type and prepends a "*".
+    // substring gets rid of the star.
+    return converter.getTypeName(service).getNickname().substring(1) + "Client";
+  }
+
+  @Override
+  public String getGrpcClientConstructorName(Interface service) {
+    return getGrpcClientTypeName(service).replace(".", ".New");
+  }
+
+  @Override
+  public String getCallOptionsTypeName(Interface service) {
+    return clientNamePrefix(service).join("call").join("options").toUpperCamel();
+  }
+
+  @Override
+  public String getDefaultClientOptionFunctionName(Interface service) {
+    return Name.lowerCamel("default")
+        .join(clientNamePrefix(service))
+        .join("client")
+        .join("options")
+        .toLowerCamel();
+  }
+
+  @Override
+  public String getDefaultCallOptionFunctionName(Interface service) {
+    return Name.lowerCamel("default")
+        .join(clientNamePrefix(service))
+        .join("call")
+        .join("options")
+        .toLowerCamel();
+  }
+
+  @Override
+  public String getClientTypeName(Interface service) {
+    return clientNamePrefix(service).join("client").toUpperCamel();
+  }
+
+  @Override
+  public String getClientConstructorName(Interface service) {
+    return Name.lowerCamel("new").join(clientNamePrefix(service)).join("client").toUpperCamel();
+  }
+
+  @Override
+  public String getPackageName() {
+    // packagePath is in form "cloud.google.com/go/library/apiv1";
+    // we want "library".
+    String[] parts = packagePath.split("/");
+    return parts[parts.length - 2];
+  }
```

Comment:
The problem is that the "library" concept in the Go world is just called "package". The "foo/bar/baz/library" that is called "package name" in ApiConfig is called "import path" in Go.

Fundamentally, there are 2 concepts in most languages and 3 in Go.
In Java:

- Declaration
  - Package name: `foo.bar`
  - Class name: `MyClass`
- Use
  - Import: `import foo.bar.MyClass`
  - Access: `MyClass.myMethod()`

In Go:

- Declaration
  - Import path: `foo/bar`
  - Package name: `baz`
    - Most of the time, this is the same with the last element of import path, in our case it's not, but that doesn't really affect this discussion
  - Class name: `MyClass`
- Use
  - Import: `import "foo/bar"`
  - Access: `baz.MyClass.MyMethod`